flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobVertexScaler.java

```diff
@@ -71,6 +71,7 @@ public JobVertexScaler(AutoScalerEventHandler<KEY, Context> autoScalerEventHandl
     public int computeScaleTargetParallelism(
             Context context,
             JobVertexID vertex,
+            boolean adjustByMaxParallelism,
```

Comment:
Can we pass-through the ShipStrategy here? That way we decide not based on some boolean but on the relevant data for making the key group adjustment decision. Also makes unit testing more solid.

Reply:
Thanks!